mkosi.default.d/fedora/10-mkosi.fedora

```diff
@@ -6,6 +6,7 @@
 [Distribution]
 Distribution=fedora
 Release=36
+KernelCommandLine=systemd.setenv=LD_PRELOAD=/usr/lib64/libasan.so.8
```

Comment:
I agree this isn't ideal. If we do it via the config file in postinst, the changes won't end up in the initrd because the postinst script runs after the initrd has been generated. We could probably put the logic in the build script instead.

I'll see if I can rework this to do that.

Reply:
Got it.

As far as I can tell arguments passed via `KernelCommandLine` aren't passed to systemd run in nspawn containers (at least with this PR applied `mkosi` failed to boot `systemd` built with ASan until `LD_PRELOAD` was added to `DefaultEnvironment` and `systemd` was reexecuted with `nsenter` manually) so it seems if there was a way to put it to config files in the build script it would automatically make it compatible with `nspawn` as well. (It could be that I screwed something up somewhere though)

Reply:
I have a PR open to make mkosi do just that. I'm also looking at adding systemd.setenv_manager to set manager environment variables from the kernel cmdline

Reply:
> I have a PR open to make mkosi pass kernel cmdline arguments to nspawn

That should work with `mkosi` but as far as I understand for example `systemd-nspawn -i` wouldn't work with those images without passing those arguments manually. I don't know If they are supposed to be used like that though.

Reply:
That's true, I thought about having systemd in nspawn read from /etc/kernel/cmdline, but that seems like it'd be a bit more of an involved change

Reply:
I think it would be possible to avoid reading `/etc/kernel/cmdline` if those images were self-contained with `LD_PRELOAD`, `ASAN_OPTIONS` and `UBSAN_OPTIONS` kept in `/etc/systemd/system.conf`. Having said that I agree that `KernelCommandLine` should be good enough at this point considering that `mkosi` is used to boot those images most of the time.

I'm wondering if there is a way to avoid setting `LD_PRELOAD` here when `systemd` is built without ASan though? My understanding is that an environment variable can be used to add `-Db_sanitize=address,undefined` and `--timeout-multiplier=...` to `mkosi.build` but I'm not sure how it can turn on/off `KernelCommandLine` conditionally.

Reply:
Yeah, that's the problem I'm running into now that prevents using KernelCommandLine. So I'm going to rework to use the config file instead to avoid that limitation since in the build script we can do an if check on the environment variable